src/utils/MainApi/MainApi.service.js

```diff
@@ -0,0 +1,74 @@
+class UserActions {
+    constructor({ url, headers }) {
+        this.url = url;
+        this.headers = headers;
+    }
+
+    async _connect(url, headers) {
+        const res = await fetch(url, headers);
+        if (res) {
```

Comment:
You need to check every response from the server if the response is `ok` or not

` if (res.ok) {` should be added


```js

    async _connect(url, headers) {
        const res = await fetch(url, headers);
        if (res.ok) {
            return res.json();
        }
        return await Promise.reject(`Error: ${res.status}`);
    }


```

Reply:
You haven't corrected ` if (res.ok) {` in `MainApi.service.js` . See my previouse remark https://github.com/AmitGit217/news-explorer-frontend/pull/2#discussion_r1017175790